spec/lending.md

```diff
@@ -1,16 +1,68 @@
+
+## The Mechanics of Lending
+
+The lending pool is made of a collection of **Reserves**, where each reserve holds liquidity of an underlying token i.e. `USDC`, `USDT` etc.. The protocol uses a tokenization strategy to achieve distribution of accrued interest. This means that each reserve is paired with a token that is given back to the lender i.e `aUSDC` and a **Lending Exchange Rate** between that token and the underlying one (`aUSDC/USDC`). The exchange rate is recalculated after every loan repayment in which it increases and in the averse case, after every loan defaults - in which it decreases. The difference in the exchange rate over the period of a year represents the annualized interest rate of lenders.
+
+Each reserve has a **Utilization Rate** associated with it. The utilization rate is the percentage of the reserve that is being currently borrowed. It changes whenever tokens are loaned, withdrawn, borrowed or repayed. Reserves also have an **Optimal Utilization Rate** which is chiefly a function of the total size of the reserve although in latter iterations it can be adjusted to represent the volatility of the reserve itself (i.e. the rate that funds flow in and out of the reserve). 
+
+Each reserve's **Interest Rate**, the percentage that borrowers must pay on their loans, is the primary tool used to curve the current utilization rate towards the optimal one. Increasing the interest rate is expected to decrease the utilization rate and vice versa. 
+
+The interest rate is fixed per borrower at the time of the loan. Therefore we are likely to observe asymmetrical sensitivity to interest changes. This means that decreasing the interest is likely to result in a relatively quick increase in the utilization rate. However the contrary, increasing the interest rate, doesn't affect current borrowers (only potential new ones) and thus we would expect a much slower decrease in utilization rate. This is something that the controlling algorithm must take into consideration.
+
+Lastly, points of crisis are when lenders try to withdraw more money than is currently available or when borrowers try to borrow more money than is available. The interest rates are used to avoid crisis by trying to achieve a safe utilization rate, however, in the event this happens the transaction will simply be denied. This means that lenders take on the risk of a) the borrowers defaulting and b) potential illiquidity of their tokens.  
+
+## Advanced Features
+
+The basics of the protocol can be furthered by a set of possible advanced features:
+
+- **Lending epochs:** To avoid hight volatility in the reserve pool, tokens that are lent could only be withdrawn at prescribed intervals or epochs. This would break the withdraw call into two calls: First, an intention to withdraw which would then set a time in the future with which the tokens become unlocked; Second, the actual withdrawal of tokens so long as the call is made after the tokens are unlocked. 
```

Comment:
```suggestion
- **Lending epochs:** To avoid high volatility in the reserve pool, tokens that are lent could only be withdrawn at prescribed intervals or epochs. This would break the withdraw call into two calls: First, an intention to withdraw which would then set a time in the future with which the tokens become unlocked; Second, the actual withdrawal of tokens so long as the call is made after the tokens are unlocked. 
```